helpers/argumentParsers.go

```diff
@@ -2,622 +2,1374 @@ package helpers
+	PR_TASK_PERF_EVENTS_ENABLE
+	PR_MCE_KILL
+	PR_MCE_KILL_GET
+	PR_SET_MM
+	PR_SET_CHILD_SUBREAPER
+	PR_GET_CHILD_SUBREAPER
+	PR_SET_NO_NEW_PRIVS
+	PR_GET_NO_NEW_PRIVS
+	PR_GET_TID_ADDRESS
+	PR_SET_THP_DISABLE
+	PR_GET_THP_DISABLE
+	PR_MPX_ENABLE_MANAGEMENT
+	PR_MPX_DISABLE_MANAGEMENT
+	PR_SET_FP_MODE
+	PR_GET_FP_MODE
+	PR_CAP_AMBIENT
+	PR_SVE_SET_VL
+	PR_SVE_GET_VL
+	PR_GET_SPECULATION_CTRL
+	PR_SET_SPECULATION_CTRL
+	PR_PAC_RESET_KEYS
+	PR_SET_TAGGED_ADDR_CTRL
+	PR_GET_TAGGED_ADDR_CTRL
+)
+
+const (
+	BPF_MAP_CREATE BPFCommandArgument = iota
+	BPF_MAP_LOOKUP_ELEM
+	BPF_MAP_UPDATE_ELEM
+	BPF_MAP_DELETE_ELEM
+	BPF_MAP_GET_NEXT_KEY
+	BPF_PROG_LOAD
+	BPF_OBJ_PIN
+	BPF_OBJ_GET
+	BPF_PROG_ATTACH
+	BPF_PROG_DETACH
+	BPF_PROG_TEST_RUN
+	BPF_PROG_GET_NEXT_ID
+	BPF_MAP_GET_NEXT_ID
+	BPF_PROG_GET_FD_BY_ID
+	BPF_MAP_GET_FD_BY_ID
+	BPF_OBJ_GET_INFO_BY_FD
+	BPF_PROG_QUERY
+	BPF_RAW_TRACEPOINT_OPEN
+	BPF_BTF_LOAD
+	BPF_BTF_GET_FD_BY_ID
+	BPF_TASK_FD_QUERY
+	BPF_MAP_LOOKUP_AND_DELETE_ELEM
+	BPF_MAP_FREEZE
+	BPF_BTF_GET_NEXT_ID
+	BPF_MAP_LOOKUP_BATCH
+	BPF_MAP_LOOKUP_AND_DELETE_BATCH
+	BPF_MAP_UPDATE_BATCH
+	BPF_MAP_DELETE_BATCH
+	BPF_LINK_CREATE
+	BPF_LINK_UPDATE
+	BPF_LINK_GET_FD_BY_ID
+	BPF_LINK_GET_NEXT_ID
+	BPF_ENABLE_STATS
+	BPF_ITER_CREATE
+	BPF_LINK_DETACH
+)
+
+const (
+	AF_UNSPEC SocketDomainArgument = iota
+	AF_UNIX
+	AF_INET
+	AF_AX25
+	AF_IPX
+	AF_APPLETALK
+	AF_NETROM
+	AF_BRIDGE
+	AF_ATMPVC
+	AF_X25
+	AF_INET6
+	AF_ROSE
+	AF_DECnet
+	AF_NETBEUI
+	AF_SECURITY
+	AF_KEY
+	AF_NETLINK
+	AF_PACKET
+	AF_ASH
+	AF_ECONET
+	AF_ATMSVC
+	AF_RDS
+	AF_SNA
+	AF_IRDA
+	AF_PPPOX
+	AF_WANPIPE
+	AF_LLC
+	AF_IB
+	AF_MPLS
+	AF_CAN
+	AF_TIPC
+	AF_BLUETOOTH
+	AF_IUCV
+	AF_RXRPC
+	AF_ISDN
+	AF_PHONET
+	AF_IEEE802154
+	AF_CAIF
+	AF_ALG
+	AF_NFC
+	AF_VSOCK
+	AF_KCM
+	AF_QIPCRTR
+	AF_SMC
+	AF_XDP
+)
+
+// OptionAreContainedInArgument checks whether the argument (rawArgument)
+// contains all of the 'options' such as with flags passed to the clone flag.
+// This function takes an arbitrary number of SystemCallArguments.It will
+// only return true if each and every option is present in rawArgument.
+// Typically linux syscalls have multiple options specified in a single
+// argument via bitmasks = which this function checks for.
+func OptionAreContainedInArgument(rawArgument uint64, options ...SystemFunctionArgument) bool {
+	var isPresent = true
+	for _, option := range options {
+		isPresent = isPresent && (option.Value()&rawArgument == option.Value())
+	}
+	return isPresent
 }
 
-// ParseMemProt parses the `prot` bitmask argument of the `mmap` syscall
-// http://man7.org/linux/man-pages/man2/mmap.2.html
-// https://elixir.bootlin.com/linux/v5.5.3/source/include/uapi/asm-generic/mman-common.h#L10
-func ParseMemProt(prot uint32) string {
+type CloneFlagArgument struct {
+	rawValue    uint64
+	stringValue string
+}
+
+func (c CloneFlagArgument) Value() uint64  { return c.rawValue }
+func (c CloneFlagArgument) String() string { return c.stringValue }
+
```

Comment:
It might be clearer if we put the CloneFlagArgument values defined above here so it is grouped in the same place.
Same for all other types.
WDYT?

Reply:
Sure, I can organize it that way